lib/pages/chat/events/video_player.dart

```diff
@@ -1,160 +1,208 @@
-                                  strokeWidth: 2,
-                                ),
-                              )
-                            : const Icon(Icons.download_outlined),
-                        label: Text(
-                          _isDownloading
-                              ? L10n.of(context)!.loadingPleaseWait
-                              : L10n.of(context)!.videoWithSize(
-                                  widget.event.sizeString ?? '?MB',
-                                ),
-                        ),
-                        onPressed: _isDownloading ? null : _downloadAction,
-                      ),
-                    ),
-                  ],
+              else
+                BlurHash(hash: blurHash),
+              Center(
+                child: ValueListenableBuilder<DownloadState>(
+                  valueListenable: _downloadStateNotifier,
+                  builder: (context, downloadState, child) {
+                    switch (downloadState) {
+                      case DownloadState.loading:
+                        return Stack(
+                          alignment: Alignment.center,
+                          children: [
+                            _CenterVideoButton(
+                              icon: Icons.play_arrow,
+                              onTap: _downloadAction,
+                            ),
+                            SizedBox(
+                              width: MessageContentStyle.videoCenterButtonSize,
+                              height: MessageContentStyle.videoCenterButtonSize,
+                              child: CircularProgressIndicator(
+                                strokeWidth: 2,
+                                color:
+                                    LinagoraRefColors.material().primary[100],
+                              ),
+                            ),
+                          ],
+                        );
+                      case DownloadState.waiting:
+                        return _CenterVideoButton(
+                          icon: Icons.play_arrow,
+                          onTap: _downloadAction,
+                        );
+                      case DownloadState.done:
+                        return _CenterVideoButton(
+                          icon: Icons.play_arrow,
+                          onTap: PlatformInfos.isWeb
+                              ? _playVideoActionWeb
+                              : _playVideoActionMobile,
+                        );
+                      case DownloadState.failed:
+                        return _CenterVideoButton(
+                          icon: Icons.error,
+                          onTap: _downloadAction,
+                        );
+                    }
+                  },
                 ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _playVideoActionWeb() async {
```

Comment:
can you export this function so I can reuse
Also I saw some where in your PR have same code, please refactor it